### PR TITLE
Module: Cosmic Bill Viewer

### PR DESCRIPTION
**Pull request by eliasgomes**

### CODE REVIEW

**Comment by eliasgomes** [comment]
go build

File: README.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
 - [Cosmic Config Server](cosmic-config-server) (powered by Spring Cloud Config Server)
 - [Cosmic Metrics Collector](cosmic-metrics-collector) (powered by Spring Boot)
 - [Cosmic Usage API](cosmic-usage-api) (powered by Spring Boot)
-- [Cosmic Usage UI](cosmic-usage-ui) (simple single-page app)
+- [Cosmic Bill Viewer](cosmic-bill-viewer) (simple web app)
```
Comment (borisroman) [inline README.md]: not power by spring boot?

File: cosmic-bill-viewer/pom.xml [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+                        <configuration>
+                            <profiles>
+                                <profile>local</profile>
+                            </profiles>
+                        </configuration>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+        <profile>
```
Comment (borisroman) [inline cosmic-bill-viewer/pom.xml:38]: Do we still want to add development?

File: cosmic-bill-viewer/src/main/resources/public/app/DetailedView.js [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+    monthSelectorComponentOnChange: function (event) {
+        event.preventDefault();
+        this.renderDomainsListLoading();
+
+        this.costCalculator = new CostCalculator(
+            this.cpuPrice,
+            this.memoryPrice,
+            this.storagePrice,
+            this.publicIpPrice,
+            this.serviceFeePercentage,
```
Comment (borisroman) [inline cosmic-bill-viewer/src/main/resources/public/app/DetailedView.js]: You're using `serviceFee` in other places.

File: cosmic-bill-viewer/src/main/resources/public/app/DetailedView.js [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+            this.innovationFeePercentage
```
Comment (borisroman) [inline cosmic-bill-viewer/src/main/resources/public/app/DetailedView.js]: Same.

File: cosmic-bill-viewer/src/main/resources/public/app/DetailedView.js [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+        this.renderDomainTableHeaders();
+        this.monthSelectorComponentOnChange(event);
+    },
+
+    parseDomainsResultDetailed: function (data) {
+        this.costCalculator.calculateDomainCosts(data.domains, true);
+        this.renderDomainsList(data.domains);
+    },
+
+    parseErrorResponse: function (response) {
```
Comment (borisroman) [inline cosmic-bill-viewer/src/main/resources/public/app/DetailedView.js:165]: Nice, finally we see the error message!

File: cosmic-bill-viewer/src/main/resources/public/app/GeneralView.js [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+    monthSelectorComponentOnChange: function (event) {
+        event.preventDefault();
+        this.renderDomainsListLoading();
+
+        this.costCalculator = new CostCalculator(
+            this.cpuPrice,
+            this.memoryPrice,
+            this.storagePrice,
+            this.publicIpPrice,
+            this.serviceFeePercentage,
```
Comment (borisroman) [inline cosmic-bill-viewer/src/main/resources/public/app/GeneralView.js]: Same, `serviceFee` is used elsewhere.

File: cosmic-bill-viewer/src/main/resources/templates/detailed.mustache [hunk 7]
```diff
@@ -478,27 +339,42 @@ <h2 id="ui-domains-table-title">Domains <small>Monthly Report</small></h2>
...
-        <script src="public/libs/underscore/underscore-min.js"></script>
-        <script src="public/app/Class.js"></script>
-        <script src="public/app/CostCalculator.js"></script>
-        <script src="public/app/Main.js"></script>
+        <script id="ui-domains-list-loading-template" type="x-tmpl-mustache">
+        {{=<% %>=}}
+            <tr class="ui-domain-no-data">
+                <td class="text-center" colspan="7">
+                    <div class="sk-wandering-cubes">
+                        <div class="sk-cube sk-cube1"></div>
```
Comment (borisroman) [inline cosmic-bill-viewer/src/main/resources/templates/detailed.mustache:347]: Beautiful.

File: cosmic-bill-viewer/src/main/resources/templates/general.mustache [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+                </div>
+                <div class="col-md-offset-1 col-md-3">
+                    <h2>Fees</h2>
+                    <h4>Service fee <span class="label label-default pull-right">{{ config.serviceFee }}%</span></h4>
+                    <h4>Innovation fee <span class="label label-default pull-right">{{ config.innovationFee }}%</span></h4>
+                </div>
+            </div>
+            <div class="row hidden-print">
+                <div id="ui-error-message" class="col-md-12">
+                    <!-- #ui-error-message-template -->
```
Comment (borisroman) [inline cosmic-bill-viewer/src/main/resources/templates/general.mustache:55]: Should there be something here?

**Review by borisroman — CHANGES_REQUESTED** [review]

File: cosmic-bill-viewer/pom.xml (hunk 2 above)
Comment (eliasgomes) [reply cosmic-bill-viewer/pom.xml:38]: Not really. Another pull request will follow to remove the `development` profile from all modules.

File: cosmic-bill-viewer/src/main/resources/templates/general.mustache (hunk 8 above)
Comment (eliasgomes) [reply cosmic-bill-viewer/src/main/resources/templates/general.mustache:55]: Nee

**Review by borisroman — APPROVED** [review]

**Pull request merged**